.github/workflows/codeql.yml

```diff
@@ -0,0 +1,93 @@
+name: "CodeQL"
+
+on:
+  push:
+    branches:
+    - main
+    - release-1[1-9].[0-9]
```

Comment:
I do not want people who backport stuff to wait until this workflow completes. Maybe I should remove this as well.

We can always trigger the test manually on any branch once this PR lands, thanks to `on.workflow_dispatch`

Reply:
The tests sometimes get too long to complete. I think having only cron jobs are enough now.